packages/find-root/src/index.ts

```diff
@@ -19,28 +19,65 @@ import {
  * monorepo implementations first, with tools based on custom file schemas
  * checked last.
  */
-const defaultOrder: Tool[] = [YarnTool, PnpmTool, LernaTool, RushTool];
+const DEFAULT_TOOLS: Tool[] = [
+  YarnTool,
+  PnpmTool,
+  LernaTool,
+  RushTool,
+  RootTool,
+];
 
 const isNoEntryError = (err: unknown): boolean =>
   !!err && typeof err === "object" && "code" in err && err.code === "ENOENT";
 
 export class NoPkgJsonFound extends Error {
+  directory: string;
+  constructor(directory: string) {
+    super(`No package.json could be found upwards from directory ${directory}`);
+    this.directory = directory;
+  }
+}
+
+export class NoMatchingMonorepoFound extends Error {
   directory: string;
   constructor(directory: string) {
     super(
-      `No package.json could be found upwards from the directory ${directory}`
+      `No monorepo matching the list of supported monorepos could be found upwards from directory ${directory}`
     );
     this.directory = directory;
   }
 }
 
-export async function findRoot(cwd: string): Promise<MonorepoRoot> {
+/**
+ * Configuration options for `findRoot` and `findRootSync` functions.
+ */
+export interface FindRootOptions {
+  /**
+   * Override the list of monorepo tool implementations that are used during the search.
+   */
+  tools?: Tool[];
+}
+
+/**
+ * Given a starting folder, search that folder and its parents until a supported monorepo
```

Comment:
We probably should stick to "directory" over "folder" since in the APIs we use `dir`, `relativeDir` etc